### PR TITLE
deprecate `misc.callInNextTick`, use `component.scheduleOnce`

### PR DESCRIPTION
**Pull request by PPpro**
Re: https://github.com/cocos/3d-tasks/issues/17767

### Changelog

* deprecate `misc.callInNextTick`, use `component.scheduleOnce`

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/asset/asset-manager/downloader.ts [hunk 1]
```diff
@@ -25,10 +25,10 @@
 import { BUILD, EDITOR, EDITOR_NOT_IN_PREVIEW } from 'internal:constants';
 import { cclegacy } from '@base/global';
 import { js } from '@base/utils';
+import { callInNextTick } from '@pal/utils';
 import { sys, misc, path } from '../../core';
 import Cache from './cache';
-import downloadFile from './download-file';
-import { FileProgressCallback } from './download-file';
+import downloadFile, { FileProgressCallback } from './download-file';
```
Comment (PPpro) [inline cocos/asset/asset-manager/downloader.ts:31]: I wanna separate these two import statement, but the current eslint `import/no-deplicates` rule has no suitable options

https://github.com/import-js/eslint-plugin-import/blob/v2.27.5/docs/rules/no-duplicates.md

File: pal/utils/src/index.ts [hunk 2]
```diff
@@ -271,6 +271,18 @@ export function clearTimeoutRAF (id: number): void {
     }
 }
 
+/**
+ * Invoke a function in next frame.
+ * @param callback The function to be invoked next frame.
+ * @param p1 The first parameter passed to `callback`.
+ * @param p2 The seconde parameter passed to `callback`.
+ */
+export function callInNextTick (callback: AnyFunction, p1?: unknown, p2?: unknown): void {
```
Comment (minggo) [inline pal/utils/src/index.ts]: Why it is part of pal?

Comment (PPpro) [reply pal/utils/src/index.ts]: because the implementation of `callInNextTick` rely on `RAF`(requestAnimationFrame) , which should be part of PAL

Comment (minggo) [reply pal/utils/src/index.ts]: The drawing process is triggered by RAF. So every codes that needs to be invoked every frame rely on RAF, for example, scheduleOnce. So i think think it is not a reasonable reason to implement `callInNextTick ` in pal.

**Review by minggo — APPROVED** [review]

**Pull request merged**